apps/builder/app/builder/features/ai/ai-fetch-result.ts

```diff
@@ -85,6 +65,31 @@ export const fetchResult = async (
     return ["Something went wrong."];
   }
 
+  const project = projectStore.get();
+  const selectedInstanceSelector = selectedInstanceSelectorStore.get();
+  const availableComponentsNames = $availableComponentsNames.get();
+  const [styles, jsx] = $jsx.get() || ["", ""];
+
+  const requestParams = {
+    jsx: `${styles}${jsx}`,
+    components: availableComponentsNames,
+    projectId: project?.id,
+    instanceId: selectedInstanceSelector?.[0],
+    prompt,
+  };
+
+  // @todo Future commands might not require all the requestParams above.
+  // When that will be the case, we should revisit the validatin below.
+  if (requestParams.instanceId === undefined) {
+    return ["You must select an element."];
+  }
+  if (
+    RequestParamsSchema.omit({ command: true }).safeParse(requestParams)
+      .success === false
+  ) {
+    return ["Invalid prompt data"];
```

Comment:
not sure when this is the case, but if its possible to get this as a user-facing issue, we might want a better message

Reply:
do you have any message suggestion for when (for whatever reason) the requestParams is not valid?

Reply:
well if we don't know anything that we can tell the user then I can't improve it